src/index.ts

```diff
@@ -67,20 +89,35 @@ export class SgidClient {
    * @param scopes Array or space-separated scopes, must include openid
    * @param nonce Specify null if no nonce
    * @param redirectUri The redirect URI used in the authorization request, defaults to the one registered with the client
+   * @param codeChallenge The code challenge from the code verifier used for PKCE enhancement
    * @returns
    */
-  authorizationUrl(
-    state: string,
-    scope: string | string[] = 'myinfo.nric_number openid',
-    nonce: string | null = generators.nonce(),
-    redirectUri: string = this.getFirstRedirectUri(),
-  ): { url: string; nonce?: string } {
+  authorizationUrl({
+    state,
+    scope = DEFAULT_SCOPE,
+    nonce = generators.nonce(),
+    redirectUri = this.getFirstRedirectUri(),
+    codeChallenge,
+  }: AuthorizationUrlParams): AuthorizationUrlReturn {
+    if (this.apiVersion !== 2) {
```

Comment:
Why do we need this line of validation if we're setting the api version ourselves?

Reply:
ah okay previously the `apiVersion` was still a parameter in the constructor with the default changed to `2`.

after removing `apiVersion` to be a constant (`API_VERSION`) that users cannot control, i forgot to remove these checks. will do so now